pg_search/src/postgres/customscan/pdbscan/exec_methods/fast_fields/mixed.rs

```diff
@@ -280,52 +257,28 @@ impl ExecMethod for MixedFastFieldExecState {
-                                .map(|att| att.name().to_string())
-                                .unwrap_or_default();
 
                             // Try the optimized fast field path first
-                            if let WhichFastField::Named(field_name, field_type) = which_fast_field
-                            {
-                                match field_type {
-                                    // String field handling
-                                    FastFieldType::String => {
-                                        if let Some(Some(term_string)) =
-                                            field_values.get_string(field_name)
-                                        {
-                                            // Use the term directly for this string field if available
-                                            if let Some(datum) =
-                                                term_to_datum(term_string, att_typid, slot)
-                                            {
-                                                datums[i] = datum;
-                                                isnull[i] = false;
-                                                continue;
-                                            }
-                                        }
-                                    }
-                                    // Numeric field handling
-                                    FastFieldType::Numeric => {
-                                        if let Some(num_value) =
-                                            field_values.get_numeric(field_name)
-                                        {
-                                            // Convert the numeric value to the right Datum type
-                                            if let Ok(Some(datum)) = TantivyValue(num_value.clone())
-                                                .try_into_datum(PgOid::from(att_typid))
-                                            {
-                                                datums[i] = datum;
-                                                isnull[i] = false;
-                                                continue;
-                                            }
-                                        }
-                                    }
+                            match field_value.try_into_datum(PgOid::from(att_typid)) {
+                                Ok(Some(datum)) => {
+                                    datums[i] = datum;
+                                    isnull[i] = false;
+                                    continue;
+                                }
+                                Ok(None) => {
+                                    // Null datum.
+                                    continue;
+                                }
+                                Err(_) => {
+                                    // TODO: panic? Is the fallthrough below actually necessary?
```

Comment:
If there's an error, we should use our error reporting mechanism, which I think is `panic` at the moment?